exporter/elasticsearchexporter/model.go

```diff
@@ -155,6 +161,131 @@ func (m *encodeModel) encodeLogECSMode(resource pcommon.Resource, record plog.Lo
 	return document
 }
 
+func (m *encodeModel) encodeMetrics(resource pcommon.Resource, metrics pmetric.MetricSlice, _ pcommon.InstrumentationScope) ([][]byte, error) {
+	hasher := sha256.New()
+	var baseDoc objmodel.Document
+
+	baseDoc.AddAttributes("", resource.Attributes())
+
+	// Put all metrics that have the same attributes and timestamp in one document.
+	docs := map[string]*objmodel.Document{}
+	for i := 0; i < metrics.Len(); i++ {
+		metric := metrics.At(i)
+
+		var dps pmetric.NumberDataPointSlice
+
+		// Only Gauge and Sum metric types are supported at the moment.
+		switch metric.Type() {
+		case pmetric.MetricTypeGauge:
+			dps = metric.Gauge().DataPoints()
+		case pmetric.MetricTypeSum:
+			dps = metric.Sum().DataPoints()
+		}
+
+		for j := 0; j < dps.Len(); j++ {
+			dp := dps.At(j)
+
+			hash := metricHash(hasher, dp.Timestamp(), dp.Attributes())
+			doc, docExists := docs[hash]
+			if !docExists {
+				doc = baseDoc.Clone()
+				doc.AddTimestamp("@timestamp", dp.Timestamp())
+				doc.AddAttributes("", dp.Attributes())
+
+				docs[hash] = doc
+			}
+
+			switch dp.ValueType() {
+			case pmetric.NumberDataPointValueTypeDouble:
+				doc.AddAttribute(metric.Name(), pcommon.NewValueDouble(dp.DoubleValue()))
+			case pmetric.NumberDataPointValueTypeInt:
+				doc.AddAttribute(metric.Name(), pcommon.NewValueInt(dp.IntValue()))
+			}
+		}
+	}
+
+	res := make([][]byte, 0, len(docs))
+
+	for _, doc := range docs {
+		if m.dedup {
+			doc.Dedup()
+		} else if m.dedot {
+			doc.Sort()
+		}
+
+		var buf bytes.Buffer
+		err := doc.Serialize(&buf, m.dedot)
+
+		if err != nil {
+			fmt.Printf("Serialize error, dropping doc: %v\n", err)
+		} else {
+			res = append(res, buf.Bytes())
+		}
+	}
+
+	return res, nil
+}
+
+func metricHash(hasher hash.Hash, timestamp pcommon.Timestamp, attributes pcommon.Map) string {
+	// Avoid any hashing if there are no attributes
+	if attributes.Len() == 0 {
+		return timestamp.String()
+	}
+
+	hasher.Reset()
+
+	hasher.Write([]byte(timestamp.AsTime().Format(time.RFC3339Nano)))
+
+	mapHash(hasher, attributes)
+
+	return string(hasher.Sum(nil))
+}
+
+func mapHash(hasher hash.Hash, m pcommon.Map) {
+	// TODO: Only hasing values, not keys? :thinking:
+	m.Range(func(_ string, v pcommon.Value) bool {
+		hasher.Write([]byte(v.Str()))
+		valueHash(hasher, v)
```

Comment:
Yes, I do think we want to hash the keys as well.
```suggestion
	m.Range(func(k string, v pcommon.Value) bool {
		hasher.Write([]byte(k.Str()))
		valueHash(hasher, v)
```

Reply:
Fixed in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33513/commits/cf4d392c28e6f67ba9cbee80fa211f887ff11a11.